### DISCUSSION TITLE
latex node for markdoc

### DISCUSSION THREAD

**Discussion by RepolloDev** [markdoc/markdoc#411] [Ideas]
In normal markdown syntax it is possible to create equations either inline or in block with the "$" symbol. For example.
This is in latex $x^2 = 4$
$$x^2=4$$
So, if possible, there could be a **latex** or **equation** node to be able to perform the same operations as with the already existing nodes.

**Comment by rpaul-stripe** [comment]
Equation syntax is a non-standard feature that isn't part of the CommonMark specification. We probably won't natively support this in Markdoc, but you can use Markdoc's extensibility to add this functionality yourself. I would suggest making an inline `{% math "..." /%}` tag for inline math and customizing the fenced code block to make it support block math.

Here's a quick-and-dirty example that I put together using `react-katex`:

```jsx
import * as Markdoc from '@markdoc/markdoc';
import * as React from 'react';
import { InlineMath, BlockMath } from 'react-katex';
import './style.css';

const components = {InlineMath, BlockMath};
const config = {
  tags: {
    math: {
      render: 'InlineMath',
      inline: true,
      attributes: {
        primary: { type: String, render: 'math', required: true },
      },
    },
  },
  nodes: {
    fence: {
      ...Markdoc.nodes.fence,
      transform(node, context) {
        return node.attributes.language === 'math' ?
          new Markdoc.Tag('BlockMath', {math: node.attributes.content.trim()}) :
          Markdoc.nodes.fence.transform(node, context);
      },
    },
  },
};


const content = String.raw`
This is an example: {% math "c = \\pm\\sqrt{a^2 + b^2}" /%}

Testing

~~~math
c = \pm\sqrt{a^2 + b^2}
~~~
`;

const doc = Markdoc.parse(content);
const transformed = Markdoc.transform(doc, config);
const output = Markdoc.renderers.react(transformed, React, {components});

export default function App() {
  return <div>{output}</div>;
}
```

Note that backslashes have to be escaped inside of strings that are passed as attribute values.
